etcdmain/config.go

```diff
@@ -127,9 +133,66 @@ type config struct {
 
 	enablePprof bool
 
+	configFile string
+
 	ignored []string
 }
 
+type yamlSecConfig struct {
+	CAFile        string `json:"ca-file"`
+	CertFile      string `json:"cert-file"`
+	KeyFile       string `json:"key-file"`
+	CertAuth      bool   `json:"client-cert-auth"`
+	TrustedCAFile string `json:"trusted-ca-file"`
+}
+
+type yamlConfig struct {
```

Comment:
@xiang90 I agree about the repetition.

Just trying to understand what you are suggesting. Currently the `config` struct has only `TickMs` and `ElectionMs` exposed and the rest of the fields are all private. You are suggesting to make all fields public except for `clusterState` and use `config` to read in the configuration file. Is my understanding correct?